File: src/main/java/no/stunor/origo/batch/data/OrganisationRepository.java
package no.stunor.origo.batch.data;

import org.springframework.stereotype.Repository;

import com.google.cloud.spring.data.firestore.FirestoreReactiveRepository;

import no.stunor.origo.batch.model.Organisation;
import reactor.core.publisher.Mono;

@Repository
public interface OrganisationRepository extends FirestoreReactiveRepository<Organisation> {
    Mono<Organisation> findByOrganisationIdAndEventorId(String organisationId, String eventorId);
    //void deleteWithLastUpdatedBefore(Timestamp timestamp);
}

File: src/main/java/no/stunor/origo/batch/data/OrganisationRepository.kt
package no.stunor.origo.batch.data

import com.google.cloud.spring.data.firestore.FirestoreReactiveRepository
import no.stunor.origo.batch.model.Organisation
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
interface OrganisationRepository : FirestoreReactiveRepository<Organisation> {
    fun findByOrganisationIdAndEventorId(organisationId: String, eventorId: String): Mono<Organisation>
    //void deleteWithLastUpdatedBefore(Timestamp timestamp);
}